### PR TITLE
py-jproperties: add v1.0.1, v2.0.0, v2.1.0, v2.1.1

### PR DESCRIPTION
**Pull request by teaguesterling**
Add a few new and older versions of the `py-jproperties` package. Older versions are needed as dependencies for upcoming packages.

### CODE REVIEW

**Comment by teaguesterling** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/py-jproperties/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    license("BSD", checked_by="teaguesterling")
+
+    version("2.1.1", sha256="40b71124e8d257e8954899a91cd2d5c0f72e0f67f1b72048a5ba264567604f29")
+    version("2.1.0", sha256="504d7b8d3b2f5f0f52c22c1f72bd50576dca17b01b4cd00d4359c6b0607a59ce")
+    version("2.0.0", sha256="b6709652f5c602e5271f519cf14cb9bf5d5a101df06e6c1d300123477a239588")
+    version("1.0.1", sha256="327e14082653a4f2212ff81a96fbf141382f727f421e8afc933bf56ff7c010f4")
+
+    depends_on("python@2.7,3:", type=("build", "run"))
+    depends_on("py-six@1.13:", type=("build", "run"))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-jproperties/package.py]: Looks like `1.0.1` doesn't have this dependency.

While `2.0.0`'s `setup.py` has:
```
    install_requires=["six ~= 1.10"],
    setup_requires=["pytest-runner ~= 2.0"],
```
but only `six` listed in the egg's `requires.txt`.  The `six` requirement should correspond to `py-six@1.10:1`.

`2.1.0`'s `setup.py` has:
```
    install_requires=["six ~= 1.12"],
    setup_requires=["pytest-runner ~= 2.0"],
```
so `six` should correspond to `py-six@1.12:1`.

Finally, `2.1.1`'s `setup.py` has the build constraints:
```
    install_requires=["six ~= 1.13"],
    setup_requires=[
        "setuptools_scm ~= 3.3"
    ],
```
which corresponds to `py-six@1.13:1` plus the additional dependency of `py-setuptools-scm@3.3:3`. Though the egg's `requires.txt` only shows the six dependency.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the homepage and version sha256s but I think the dependencies for the versions need a bit more work.

File: var/spack/repos/builtin/packages/py-jproperties/package.py (hunk 1 above)
Comment (teaguesterling) [reply var/spack/repos/builtin/packages/py-jproperties/package.py]: Thank you! Will add these in. I really appreciate the time to capture this. It helps me think through what to look for in future PRs.

**Review by tldahlgren — APPROVED** [review]
Thanks for the updates.

**Pull request merged**